### PR TITLE
ZTS: Move dumpdev tests to sunos.run

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

Neither FreeBSD nor Linux support dumping to zvols.

### Description

* Remove tests for using zvols as a dump device from common.run and add them to sunos.run
* Remove FreeBSD adaptations from dumpdev helper functions.
* Remove exceptions for these tests from zts-report.
* An `is_illumos` function was added to libtest.shlib to eliminate some awkward platform checks

### How Has This Been Tested?

Passed ZTS

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by ikozhukhov** [comment]
we are using there tests on DilOS - please do not remove it, you can just disable it

**Comment by freqlabs** [comment]
> we are using there tests on DilOS - please do not remove it, you can just disable it

Ok, I've put the files back and only removed them from the runfile and the exceptions from zts-report.

File: tests/runfiles/common.run [hunk 1]
```diff
@@ -879,10 +879,8 @@ tests = ['zvol_cli_001_pos', 'zvol_cli_002_pos', 'zvol_cli_003_neg']
 tags = ['functional', 'zvol', 'zvol_cli']
 
 [tests/functional/zvol/zvol_misc]
-tests = ['zvol_misc_001_neg', 'zvol_misc_002_pos', 'zvol_misc_003_neg',
-    'zvol_misc_004_pos', 'zvol_misc_005_neg', 'zvol_misc_006_pos',
-    'zvol_misc_hierarchy', 'zvol_misc_rename_inuse', 'zvol_misc_snapdev',
-    'zvol_misc_volmode', 'zvol_misc_zil']
+tests = ['zvol_misc_002_pos', 'zvol_misc_hierarchy', 'zvol_misc_rename_inuse',
+    'zvol_misc_snapdev', 'zvol_misc_volmode', 'zvol_misc_zil']
```
Comment (behlendorf) [inline tests/runfiles/common.run]: As a matter of documentation, let's make it clear why these still exist in the repository but are not referenced by the `common.run` file by adding them to an `illumos.run` file.

Comment (freqlabs) [reply tests/runfiles/common.run]: @ikozhukhov can you confirm the output of `uname` on DilOS is "illumos"?

Comment (freqlabs) [reply tests/runfiles/common.run]: I checked a DilOS installer in a VM and the uname is "SunOS" so I will name the file `sunos.run`

Comment (freqlabs) [reply tests/runfiles/common.run]: Actually `uname -o` reports "illumos" so I'll update things to check that instead.

Comment (freqlabs) [reply tests/runfiles/common.run]: Nevermind, that would be a problem with "GNU/Linux". It's simplest to just name the file `sunos.run` in this case.

**Comment by freqlabs** [comment]
* Move tests using dumpadm to sunos.run
* Remove unnecessary platform checks from relocated tests
* Add `is_illumos` function to simplify some checks

**Comment by freqlabs** [comment]
* Simplify a few more checks

File: tests/runfiles/common.run (hunk 1 above)
Comment (ikozhukhov) [reply tests/runfiles/common.run]: it'll be `dilos.run` in future :)

Comment (ikozhukhov) [reply tests/runfiles/common.run]: i'll add later:
https://bitbucket.org/dilos/dilos-illumos/src/87795b2e11801e17c89487232355c16264e2bdcc/usr/src/test/zfs-tests/include/libtest.shlib#lines-96
when Brian will give me approve :)

Comment (behlendorf) [reply tests/runfiles/common.run]: Sounds good.  Then for the moment, let's group these `sunos.run` as done in this PR.

**Pull request merged**